scripts/us_census/acs5yr/subject_tables/common/acs_spec_validator.py

```diff
@@ -0,0 +1,849 @@
+    """Find list of tokens that appear in ignoreColumn as well as a lookup for PV
+    NOTE: checks only tokens, ignores long column names
+
+    Args:
+      spec_dict: Dict obj containing configurations for the import.
+      delimiter: delimiter seperating tokens within single column name string.
+    
+    Returns:
+      List of tokens that are part of ignoreColumns as well as pv section.
+  """
+    ret_list = []
+
+    new_dict = copy.deepcopy(spec_dict)
+    new_dict.pop('ignoreColumns', None)
+    new_dict.pop('ignoreTokens', None)
+
+    spec_tokens = get_spec_token_list(new_dict, delimiter)['token_list']
+
+    if 'ignoreColumns' in spec_dict:
+        for ignore_token in spec_dict['ignoreColumns']:
+            if ignore_token in spec_tokens:
+                ret_list.append(ignore_token)
+
+    if 'ignoreTokens' in spec_dict:
+        for ignore_token in spec_dict['ignoreTokens']:
+            if ignore_token in spec_tokens:
+                ret_list.append(ignore_token)
+
+    return ret_list
+
+
+def find_missing_enum_specialisation(column_name_list: list,
+                                     spec_dict: dict,
+                                     delimiter: str = '!!') -> dict:
+    """Check for missing entries in the enumSpecializations section of the spec
+      If multiple tokens match same property, they should appear as enumspecialisation
+      the token that appears later in the name should be the specialisation of one one encountered before
+    
+    Args:
+      column_name_list: List of all columns after dropping ignored columns.
+      spec_dict: Dict obj containing configurations for the import.
+      delimiter: delimiter seperating tokens within single column name string.
+    
+    Returns:
+      Dictionary with lookup token as key value. Each token has a dict associated with following keys:
+        - column: List of columns where the token appears.
+        - possibleParents: List of possible values of the property that might be replacable.
+  """
+    ret_dict = {}
+    for column_name in column_name_list:
+        temp_dict = {}
+        # populate a dictionary containing properties and all the values assigned to it
+        for token in column_name.split(delimiter):
+            for prop in spec_dict['pvs'].keys():
+                if token_in_list_ignore_case(token,
+                                             spec_dict['pvs'][prop].keys()):
+                    if prop in temp_dict:
+                        temp_dict[prop].append(token)
+                    else:
+                        temp_dict[prop] = [token]
+        # check all the columns that have multiple values assigned to a single property
+        for prop in temp_dict:
+            if len(temp_dict[prop]) > 1:
+                # retDict.append(tempDict[prop])
+                for i, prop_token in enumerate(reversed(temp_dict[prop])):
+                    j = len(temp_dict[prop]) - 1 - i
```

Comment:
Can do reversed after enumerate:
`for j , prop_token in reversed(list(enumerate(temp_dict[prop]))):`

Reply:
Thanks for the suggestion. Fixed.